src/main/java/com/example/petinnoflightandairlines/service/AirportServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package com.example.petinnoflightandairlines.service;
+
+import com.example.petinnoflightandairlines.dto.AirportDTO;
+import com.example.petinnoflightandairlines.mapper.AirportMapper;
+import com.example.petinnoflightandairlines.model.Airport;
+import com.example.petinnoflightandairlines.repository.AirportRepository;
+import io.micrometer.common.util.StringUtils;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class AirportServiceImpl implements AirportService {
+
+    private final AirportRepository airportRepository;
+
+    private final AirportMapper airportMapper;
+
+    @Override
+    public AirportDTO addAirport(@Valid AirportDTO airportDTO) {
+
+        checkAirportIata(airportDTO);
+        Airport airport = airportMapper.convertAirportDTOToAirport(airportDTO);
+
+        Airport savedAirport = saveAirport(airport);
+        return airportMapper.convertAirportToAirportDTO(savedAirport);
+    }
+
+    @Override
+    public AirportDTO getAirportDTO(Long airportId) {
```

Comment:
1. getAirportDTO -> getAirportById
2. better to use 1 method search with all filters